Source/Details/ASElementMap.m

```diff
@@ -133,6 +133,17 @@ - (void)enumerateUsingBlock:(void(^)(NSIndexPath *indexPath, ASCollectionElement
   }];
 }
 
+- (NSArray<ASCollectionElement *> *)filteredElementsUsingBlock:(BOOL (^)(NSIndexPath * _Nonnull, ASCollectionElement * _Nonnull, BOOL * _Nonnull))filteringBlock
+{
+  NSMutableArray<ASCollectionElement *> *elements = [NSMutableArray array];
+  [self enumerateUsingBlock:^(NSIndexPath * _Nonnull indexPath, ASCollectionElement * _Nonnull element, BOOL * _Nonnull stop) {
```

Comment:
Would be great to avoid block-based enumeration.  Especially in our highly concurrent system, these retains and releases are the #1 most common impact on performance regressions & wins — this is notable because not only do they show up prominently in time profiles, but especially the system trace shows the runtime lock contention is the biggest cause of incomplete CPU utilization.

cc @Adlai-Holler @maicki to think of ways we can solve this more systematically (if open to the idea, I think a code style with stuff like this _specifically for core code_ could be one step).

This should _obviously_ not block this diff from landing; however I think it is justified to file a followup task now if we'd like to defer addressing it.  We can spend the time to profile it, but it is a time investment to do the system tracing necessary to understand the impacts.